massa-execution-worker/src/execution.rs

```diff
@@ -864,9 +764,33 @@ impl ExecutionState {
         Ok(context_guard!(self).settle_slot())
     }
 
+    /// Check that the slot is within the reach of history
+    fn verify_active_slot(&self, slot: Slot) {
+        if slot <= self.final_cursor {
+            panic!("cannot execute at a slot before finality");
+        }
+        let max_slot = self
+            .active_cursor
+            .get_next_slot(self.config.thread_count)
+            .expect("slot overflow when getting speculative ledger");
+        if slot > max_slot {
+            panic!("cannot execute at a slot beyond active cursor + 1");
+        }
+    }
+
+    /// Computes the index of a given slot in the active history
+    fn get_active_index(&self, slot: Slot) -> Option<usize> {
+        if let Some(hist_front) = &self.active_history.read().0.front() {
+            slot.slots_since(&hist_front.slot, self.config.thread_count)
```

Comment:
it is worth checking that the returned index is `>= active_history.len()` and returning None if it is the case

Reply:
Done 👍